File: app/src/main/java/com/mahadum360/mahadum/ui/activities/auth/di/AuthActivityContextModule.java
package com.mahadum360.mahadum.ui.activities.auth.di;

import com.mahadum360.mahadum.data.ApiService;
import com.mahadum360.mahadum.data.RealmService;
import com.mahadum360.mahadum.ui.activities.auth.AuthContract;
import com.mahadum360.mahadum.ui.activities.auth.AuthPresenter;

import dagger.Module;
import dagger.Provides;

@Module
public class AuthActivityContextModule {


    private AuthContract.View view;

    public AuthActivityContextModule(AuthContract.View view) {
        this.view = view;
    }

    @Provides
    @AuthScope
    AuthContract.View providesView() {
        return view;
    }

    @Provides
    @AuthScope
    AuthPresenter providesPresenter(ApiService apiService, RealmService realmService, AuthContract.View view) {
        return new AuthPresenter(apiService, realmService, view);
    }
}

File: app/src/main/java/com/mahadum360/mahadum/auth/di/AuthActivityContextModule.kt
package com.mahadum360.mahadum.auth.di

import com.mahadum360.mahadum.data.ApiService
import com.mahadum360.mahadum.data.RealmService
import com.mahadum360.mahadum.auth.AuthContract
import com.mahadum360.mahadum.auth.AuthPresenter
import com.mahadum360.mahadum.utils.schedule.SchedulerProvider

import dagger.Module
import dagger.Provides

@Module
class AuthActivityContextModule(private val view: AuthContract.View) {

    @Provides
    @AuthScope
    internal fun providesView(): AuthContract.View {
        return view
    }

    @Provides
    @AuthScope
    internal fun providesPresenter(apiService: ApiService, realmService: RealmService, view: AuthContract.View,
                                   provider: SchedulerProvider): AuthPresenter {
        return AuthPresenter(apiService, realmService, view, provider)
    }
}
